### ISSUE TITLE
intro-neural-networks/gradient-descent The error being calculated per example but being updated by epoch

### ISSUE THREAD

**Issue by Blowoffvalve** [udacity/deep-learning-v2-pytorch#125]
In the train method of the gradient-descent notebook, the error is being computed for every sample in the training set(making it seem like you want to do Stochastic Gradient Descent with that error), however the error isn't being used. Instead, the error is being computed again inside update_weights(). If you accept the issue, i can submit a pull request for the fix.

```    
for e in range(epochs):
        del_w = np.zeros(weights.shape)
        for x, y in zip(features, targets):
            output = output_formula(x, weights, bias)
            #This is where the error is. You can safely remove the line below.
            error = error_formula(y, output)
            weights, bias = update_weights(x, y, weights, bias, learnrate)
        
        # Printing out the log-loss error on the training set
        out = output_formula(features, weights, bias)
        loss = np.mean(error_formula(targets, out))
        errors.append(loss)```

**Comment by hoxiea** [comment]
Good catch, @Blowoffvalve . `del_w` doesn't get used in this code either, because the update is happening directly in `update_weights`. I suggested removing that line in [this pull request](https://github.com/udacity/deep-learning-v2-pytorch/pull/140).

**Comment by jchernus** [comment]
Thanks folks! 

@Blowoffvalve, what solution do you propose? I'll hold off on accepting @hoxiea's PR until I hear back from @Blowoffvalve.

**Comment by Blowoffvalve** [comment]
Hello @jchernus , I think we can accept @hoxiea 's PR.

**Comment by abhiojha8** [comment]
Closing this due to no response from the reporter.
